iXBRLViewerPlugin/viewer/src/js/calculation.js

```diff
@@ -0,0 +1,202 @@
+// See COPYRIGHT.md for copyright information
+
+import $ from 'jquery';
+import Decimal from 'decimal.js';
+import { setDefault } from './util.js';
+import { Interval } from './interval.js';
+import { FactSet } from './factset.js';
+
+export class Calculation {
+    
+    constructor(fact, calc11) {
+        this.fact = fact;
+        this.calc11 = calc11;
+    }
+
+    /* Resolve calculation relationships to a map of maps of maps 
+     * (ELR->conceptName->fact id->fact object) */
+    calculationFacts() {
+        const fact = this.fact;
+        const report = fact.report;
+        if (!this._conceptToFact) {
+            const rels = report.getChildRelationships(fact.conceptName(), "calc")
+            const ctf = {};
+            for (const [elr, rr] of Object.entries(rels)) {
+                ctf[elr] = {};
```

Comment:
Each value in `ctf` is a map of concept name to a set of facts.  We can definitely end up with multiple facts for the same concept, and we want to be able to look them up easily by concept name.

What I'm less sure of is why I've got facts as a map of fact ID to fact object.  I don't think we ever look up facts in this map by ID.

Reply:
If you have `sumA->[elemB,elemB]` with `factB1,factB`, will we attempt to process concept B twice?  In that case, an object would dedup to `{b1:factB1,b2:factB2}`, but an array would be `[factB1,factB2,factB1,factB2]`.  If that's right, an object for dedup seems good, so I was mostly just thinking out loud about my understanding.

Reply:
That is true.  That arrangement illegal in Calc 1.1 and of questionable legality in Calc 1.0 (https://gitlab.xbrl.org/base-spec/calculations/-/issues/13) but we should probably handle it more cleanly.

I might try making `FactSet` actually behave like a set and de-duplicate facts, as we could then assemble the `FactSet` here rather than later.

Reply:
Done in 21b35f3